src/pages/auth/login.js

```diff
@@ -1,38 +1,20 @@
-import axios from "axios";
 import { Link, useNavigate } from "react-router-dom";
 import { useState } from "react";
-import { useProductList } from "../../context/productListContext";
 import { User } from "../../context/userContext";
-import { USER_LOADING, LOGIN_STATUS } from "../../reducer/user/userConstants";
+import { login } from "../../utils/handler";
 
 export const LogIn = () => {
   const [credentails, setCredentials] = useState({});
   const { state, dispatch } = User();
 
   let navigate = useNavigate();
 
-  const login = async (email, password) => {
-    try {
-      const response = await axios.post("/api/auth/login", {
-        email,
-        password,
-      });
-      localStorage.setItem("token", response.data.encodedToken);
-      navigate("/");
-      dispatch({ type: USER_LOADING });
-      dispatch({type: LOGIN_STATUS})
-      setTimeout(() => dispatch({ type: USER_LOADING }), 1000);
-    } catch (err) {
-      console.log("Error: ", err);
-    }
-  };
-
   const loginHandler = async () => {
-    await login(credentails.email, credentails.password);
+    await login(credentails.email, credentails.password, dispatch, navigate);
```

Comment:
pass whole credentials as object credentials which makes params shorter to read and work